pkg/operator/controllers/muo/muo_controller.go

```diff
@@ -66,7 +86,48 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	// If enabled and managed=false, remove the MUO deployment
 	// If enabled and managed is missing, do nothing
 	if strings.EqualFold(managed, "true") {
-		err = r.deployer.CreateOrUpdate(ctx, instance)
+		imagePullspec, ext := instance.Spec.OperatorFlags[controllerPullSpec]
+		if !ext {
+			return reconcile.Result{}, errors.New("missing pullspec")
+		}
+
+		config := &config.MUODeploymentConfig{
+			Pullspec: imagePullspec,
+		}
+
+		allowOCM := instance.Spec.OperatorFlags.GetSimpleBoolean(controllerAllowOCM)
+		if allowOCM {
+			useOCM := func() bool {
+				var userSecret *corev1.Secret
+
+				userSecret, err = r.kubernetescli.CoreV1().Secrets(pullSecretName.Namespace).Get(ctx, pullSecretName.Name, metav1.GetOptions{})
+				if err != nil {
+					// if a pullsecret doesn't exist/etc, fallback to local
+					return false
+				}
+
+				parsedKeys, err := pullsecret.UnmarshalSecretData(userSecret)
+				if err != nil {
+					// if we can't parse the pullsecret, fallback to local
+					return false
+				}
+
+				// check for the key that connects the cluster to OCM (since
+				// clusters may have a RH registry pull secret but not the OCM
+				// one if they choose)
+				_, foundKey := parsedKeys[pullSecretOCMKey]
+				return foundKey
+			}()
```

Comment:
We already expose `cloud.redhat.com` in the status field of `cluster.aro.openshift.io` custom resource, but it makes sense to keep this here in case the url changes, the pull-secret controller errors out, etc.

Reply:
Yeah, that was my thinking.